src/msemu.c

```diff
@@ -132,22 +123,22 @@ void writeLCD(ms_ctx* ms, uint16_t newaddr, uint8_t val, int lcdnum)
 //
 uint8_t readLCD(ms_ctx* ms, uint16_t newaddr, int lcdnum)
 {
-	uint8_t *lcd_ptr;
+	uint32_t *lcd_ptr;
 	uint8_t ret;
 
-	if (lcdnum == LCD_L) lcd_ptr = ms->lcd_dat1bit;
+	if (lcdnum == LCD_L) lcd_ptr = ms->lcd_datRGBA8888;
 	/* XXX: Fix the use of this magic number, replace with a const */
-	else lcd_ptr = &ms->lcd_dat1bit[4800];
+	else lcd_ptr = &ms->lcd_datRGBA8888[4800];
 
 	/* XXX: This might need to be reworked to use non-viewable LCD memory */
 	// Wraps memory address if out of bounds
-	if (newaddr >= 240) newaddr %= 240;
+	if (newaddr >= MS_LCD_HEIGHT) newaddr %= MS_LCD_HEIGHT;
 
 	// Check CAS bit on P2
 	if (ms->io[MISC2] & 8)
 	{
 		// Return data on currently selected LCD column
-		ret = lcd_ptr[newaddr + (ms->lcd_cas * 240)];
+		ret = lcd_ptr[newaddr + (ms->lcd_cas * MS_LCD_HEIGHT)];
```

Comment:
This will overflow. I don't think this will convert RGBA8888 into a 1 bit representation that the mailstation expects consistently.

If it does, will it continue to work if we change the colors?